Sources/Filters/Sources/PlatonicSolidSource/index.js

```diff
@@ -0,0 +1,197 @@
+      16, 8, 15, 5, 4, 17, 10, 12, 6, 7, 18, 11, 14, 7, 6, 19, 9, 13, 8, 16, 0,
+      13, 9, 9, 19, 2, 15, 8, 10, 17, 3, 14, 11, 11, 18, 1, 12, 10,
+    ],
+    numPoints: 20,
+    cellSize: 5,
+    numCells: 12,
+    scale: 1.0 / 1.070466269319,
+  },
+};
+
+// ----------------------------------------------------------------------------
+// vtkPlatonicSolidSource methods
+// ----------------------------------------------------------------------------
+
+function vtkPlatonicSolidSource(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkPlatonicSolidSource');
+
+  publicAPI.requestData = (inData, outData) => {
+    const output = outData[0] || vtkPolyData.newInstance();
+
+    let solidData;
+    switch (model.solidType) {
+      case SolidType.VTK_SOLID_TETRAHEDRON:
+        solidData = geometries.tetrahedron;
+        break;
+      case SolidType.VTK_SOLID_CUBE:
+        solidData = geometries.cube;
+        break;
+      case SolidType.VTK_SOLID_OCTAHEDRON:
+        solidData = geometries.octahedron;
+        break;
+      case SolidType.VTK_SOLID_ICOSAHEDRON:
+        solidData = geometries.icosahedron;
+        break;
+      case SolidType.VTK_SOLID_DODECAHEDRON:
+        solidData = geometries.dodecahedron;
+        break;
+      default:
+        solidData = geometries.tetrahedron;
+        break;
+    }
+
+    let pointType;
+    if (model.outputPointsPrecision === DesiredOutputPrecision.SINGLE) {
+      pointType = VtkDataTypes.FLOAT;
+    } else if (model.outputPointsPrecision === DesiredOutputPrecision.DOUBLE) {
+      pointType = VtkDataTypes.DOUBLE;
+    }
+
+    const points = vtkPoints.newInstance({
+      dataType: pointType,
+      numberOfPoints: solidData.numPoints,
+    });
+
+    const colors = vtkDataArray.newInstance({
+      name: 'Colors',
+      numberOfComponents: 1,
+      values: new Uint32Array(solidData.numCells),
+    });
+
+    for (let i = 0; i < solidData.points.length; i += 3) {
+      points.insertNextPoint(
+        solidData.scale * solidData.points[i],
+        solidData.scale * solidData.points[i + 1],
+        solidData.scale * solidData.points[i + 2]
+      );
+    }
+
+    const polys = vtkCellArray.newInstance();
+    for (let i = 0; i < solidData.cells.length; i += solidData.cellSize) {
+      const cell = solidData.cells.slice(i, i + solidData.cellSize);
+      polys.insertNextCell(cell);
+    }
+
+    for (let i = 0; i < solidData.numCells; i++) {
+      colors.setTuple(i, [i]);
+    }
+
+    output.setPoints(points);
+    output.setPolys(polys);
+    output.getCellData().setScalars(colors);
+
+    outData[0] = output;
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  solidType: SolidType.VTK_SOLID_TETRAHEDRON,
+  outputPointsPrecision: DesiredOutputPrecision.DEFAULT,
```

Comment:
you might want to add a global "scale" parameter

Reply:
You mean replace the scale per solid with a global one ?

Reply:
the end user might want to have a cube of size 10, how to easily do that ?